File: 0268.cpp
// Link: https://leetcode.com/problems/missing-number/
// Runtime: 14 ms 
// Memory: 18.1 MB

class Solution 
{
public:
    int missingNumber(const std::vector<int>& nums) 
    {
        int totalSum = nums.size() * (nums.size() + 1) / 2;
        for (int value : nums)
            totalSum -= value;
        return totalSum;
    }
};

File: python/0268.py
# Link: https://leetcode.com/problems/missing-number/description/
# Runtime: 109 ms
# Memory: 17.81 MB

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        sum_of_array = len(nums) * (len(nums) + 1) // 2
        for num in nums:
            sum_of_array -= num
        
        return sum_of_array
